File: src/main/java/com/wxl/dyttcrawler/utils/DateUtils.java
package com.wxl.dyttcrawler.utils;

import java.time.LocalDate;
import java.time.ZoneId;
import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;
import java.util.Date;

/**
 * Create by wuxingle on 2020/7/18
 * 日期工具类
 */
public class DateUtils {

    /**
     * 解析为Date类型
     */
    public static Date parseDate(String dateStr, DateTimeFormatter formatter)
            throws DateTimeParseException {
        ZonedDateTime dateTime = LocalDate.parse(dateStr, formatter).atStartOfDay(ZoneId.systemDefault());
        return Date.from(dateTime.toInstant());
    }


}

File: src/main/kotlin/com/wxl/dyttcrawler/utils/DateUtils.kt
package com.wxl.dyttcrawler.utils

import java.time.LocalDate
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeParseException
import java.util.*

/**
 * Create by wuxingle on 2021/10/11
 * 日期工具类
 */
class DateUtils {

    /**
     * 解析为Date类型
     */
    companion object {

        fun parseDate(dateStr: String, formatter: DateTimeFormatter): Date {
            val dateTime = LocalDate.parse(dateStr, formatter).atStartOfDay(ZoneId.systemDefault())
            return Date.from(dateTime.toInstant())
        }

        fun parseDate(dateStr: String, vararg formatters: DateTimeFormatter): Date? {
            for (formatter in formatters) {
                try {
                    val dateTime = LocalDate.parse(dateStr, formatter).atStartOfDay(ZoneId.systemDefault())
                    return Date.from(dateTime.toInstant())
                } catch (e: DateTimeParseException) {
                    //ignore
                }
            }
            return null
        }
    }


}
